### PR TITLE
Added DL cases for Javascript

### PR DESCRIPTION
**Pull request by aastjais**
Added compliant and noncompliant DL case for Javascript.

Note: All the test cases have 100% recall and precision.

### CODE REVIEW

**Review by saurabh-amazon — APPROVED** [review]
Reviewed all changes. Changes look fine. Approved.

File: javascript/src/detectors/javascript-avoid-nan-in-comparison/javascript-avoid-nan-in-comparison_compliant.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+// {fact rule=javascript-avoid-nan-in-comparison@v1.0 defects=0}
+for (let i = 0; i < cars.length; i++) {
+    console.log(cars[i])
+    // Compliant : `isNaN()` function from Number will check for NaN value from array.
```
Comment (grl-wang) [inline javascript/src/detectors/javascript-avoid-nan-in-comparison/javascript-avoid-nan-in-comparison_compliant.js:7]: could you pls change to `// Compliant: `

Comment (grl-wang) [reply javascript/src/detectors/javascript-avoid-nan-in-comparison/javascript-avoid-nan-in-comparison_compliant.js:7]: we have strict compliant comment format validation in lib publish system. could you pls make sure your example follow the code example formate

File: javascript/src/detectors/javascript-avoid-nan-in-comparison/javascript-avoid-nan-in-comparison_non-compliant.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+// {fact rule=javascript-avoid-nan-in-comparison@v1.0 defects=1}
+const cars = ["Saab", "Volvo", "BMW", NaN];
+
+for (let i = 0; i < cars.length; i++) {
+    // Noncompliant : `isNaN()` is not used.
```
Comment (grl-wang) [inline javascript/src/detectors/javascript-avoid-nan-in-comparison/javascript-avoid-nan-in-comparison_non-compliant.js:8]: pls update the comment format

**Review by saurabh-amazon — APPROVED** [review]
Reviewed corrections. Corrections look ok. Approved.

**Review by grl-wang — APPROVED** [review]

**Pull request merged**